src/SignalRServiceExtension/Client/AzureSignalRClient.cs

```diff
@@ -65,13 +66,68 @@ public Task SendToUser(string hubName, string userId, SignalRData data)
+            var userIdsSegment = string.IsNullOrEmpty(Version) ? $"/user/{userId}" : $"/users/{userId}";
             var connectionInfo = GetServerConnectionInfo(hubName, userIdsSegment);
             var uri = $"{connectionInfo.Url}{userIdsSegment}";
-            return PostJsonAsync(uri, data, connectionInfo.AccessToken);
+            return RequestAsync(uri, data, connectionInfo.AccessToken, HttpMethod.Post);
         }
 
-        private (string EndPoint, string AccessKey) ParseConnectionString(string connectionString)
+        public Task SendToGroup(string hubName, string groupName, SignalRData data)
+        {
+            if (string.IsNullOrEmpty(groupName))
+            {
+                throw new ArgumentException($"{nameof(groupName)} cannot be null or empty");
+            }
+
+            var groupSegment = string.IsNullOrEmpty(Version) ? $"/group/{groupName}" : $"/groups/{groupName}";
+            var connectionInfo = GetServerConnectionInfo(hubName, groupSegment);
+            var uri = $"{connectionInfo.Url}{groupSegment}";
+            return RequestAsync(uri, data, connectionInfo.AccessToken, HttpMethod.Post);
+        }
+
+        public Task AddUser(string hubName, string userId, string group)
+        {
+            if (string.IsNullOrEmpty(Version))
+            {
+                throw new ArgumentException($"API AddUserToGroup is support after Version = 1.0, check SignalR connection string to verify Version information");
+            }
+            if (string.IsNullOrEmpty(userId))
+            {
+                throw new ArgumentException($"{nameof(userId)} cannot be null or empty");
+            }
+            if (string.IsNullOrEmpty(group))
+            {
+                throw new ArgumentException($"{nameof(group)} cannot be null or empty");
+            }
+
+            var userGroupSegment = $"/groups/{group}/users/{userId}";
+            var connectionInfo = GetServerConnectionInfo(hubName, userGroupSegment);
+            var uri = $"{connectionInfo.Url}{userGroupSegment}";
+            return RequestAsync(uri, null, connectionInfo.AccessToken, HttpMethod.Put);
+        }
+
+        public Task RemoveUser(string hubName, string userId, string group)
+        {
+            if (string.IsNullOrEmpty(Version))
+            {
+                throw new ArgumentException($"API AddUserToGroup is support after Version = 1.0, check SignalR connection string to verify Version information");
+            }
+            if (string.IsNullOrEmpty(userId))
+            {
+                throw new ArgumentException($"{nameof(userId)} cannot be null or empty");
+            }
+            if (string.IsNullOrEmpty(group))
+            {
+                throw new ArgumentException($"{nameof(group)} cannot be null or empty");
+            }
+
+            var userGroupSegment = $"/groups/{group}/users/{userId}";
+            var connectionInfo = GetServerConnectionInfo(hubName, userGroupSegment);
```

Comment:
Since you already pass in additionalPath, return server url with additionalPath so you don't need to append it again?

Reply:
Good catch!